test/FullRange.t.sol

```diff
@@ -90,8 +88,12 @@ contract TestFullRange is Test, Deployers, GasSnapshot {
 
         manager = new PoolManager(500000);
 
-        FullRangeImplementation impl = new FullRangeImplementation(manager, fullRange);
-        vm.etch(address(fullRange), address(impl).code);
+        uint160 flags =
+            uint160(Hooks.BEFORE_INITIALIZE_FLAG | Hooks.BEFORE_MODIFY_POSITION_FLAG | Hooks.BEFORE_SWAP_FLAG);
+        (address hookAddress, bytes32 salt) =
+            HookMiner.find(address(this), flags, 0, type(FullRange).creationCode, abi.encode(manager));
+        fullRange = new FullRange{salt: salt}(manager);
+        require(address(fullRange) == hookAddress, "TestFullRange: hook address mismatch");
```

Comment:
confused about this check. Is there a scenario where this would be false and we'd want to halt the test? Doesn't the check on line 32 in HookMiner take care of this? Feels extra, or like we don't trust our hook miner...in which case it could maybe have its own tests 🤔

Reply:
It wouldnt fail and we can trust the miner. It's really a safety check since there's a minor footgun with the `deployer` argument in `HookMiner.find(address deployer, ...)`

In tests, `deployer` can be `address(this)` or the pranking address from `vm.prank`. There's a probability that the mined salt creates a valid address with the wrong deployer. i.e. `HookMiner.find(alice, ...)` but forgetting to prank on `new FullRange`. This would cause a mismatch between `hookAddress` and `address(fullRange)`

The hook would still deploy but a dev might use `hookAddress` in a way that causes false positive bugs/errors.

Happy to remove it or change it to `assertEq` (to imply a safety check)

Reply:
after thinking about it some more, decided to remove it entirely. while it is good to encourage safety checks, i dont think its something that had to be showcased in the codebase